### DISCUSSION TITLE
Dynamically modify a form window

### DISCUSSION THREAD

**Discussion by sab4tg** [Vector35/binaryninja-api#1961] [General] [answered]
Apologize in advance if this has been answered elsewhere, I wasn't able to find an answer to my question.

I'd like to know how to present the user with a form window that dynamically loads an extra text input field depending on the users input to a previous input field. So for example, if the users response to input_field_1 is "Other", the second input field is generated within the existing form window allowing the user to provide text input which can then be handled accordingly. Is there an easy way to do this via the binaryninja API, or would this require JS? Any ideas or thoughts help.

For context, I'm writing a Python plugin and am currently using the ```get_form_input```, ```ChoiceField```, and ```TextLineField``` functions for my form window, permanent input field, and dynamic input field, respectfully.

**Comment by plafosse** [comment, marked as answer]
So this is much more complicated than Binary Ninja's Core's simple "interaction" API allows.  You would need to write some QT code and use the binaryninjaui API. There are some examples of using this in the examples directory.

**Reply by sab4tg** [reply]
@plafosse Thank you for your reply! Could you ellaborate which example demonstrates this functionality? I couldn't find any Python plugins that exhibit this functionality. Thanks again for your help!

**Reply by psifertex** [reply]
Sure! Kaitai, Snippets, and Triage all demonstrate different UI plugins that use PySide2 to extend the UI. They're the top three folders here:

https://github.com/Vector35/binaryninja-api/tree/dev/python/examples

Some other examples include:
https://github.com/gaasedelen/lighthouse
https://gist.github.com/psifertex/6fbc7532f536775194edd26290892ef7#file-simple_ui-py
